models/DataObject/Service.php

```diff
@@ -432,11 +432,12 @@ public static function gridObjectData(AbstractObject $object, array $fields = nu
                                     }
                                 }
                             } else {
+                                $inheritanceBackup = AbstractObject::getGetInheritedValues();
                                 AbstractObject::setGetInheritedValues(true);
 
-                                $data[$dataKey] = $object->get($key, $user->getLanguage());
+                                $data[$dataKey] = $object->get($key, $user?->getLanguage());
 
-                                AbstractObject::setGetInheritedValues(false);
+                                AbstractObject::setGetInheritedValues($inheritanceBackup);
```

Comment:
recommending to use `useInheritedValues` , if that's the same intended behavior https://github.com/pimcore/pimcore/blob/79e57f164c90fc33af46c3f6de26cda387b722fd/models/DataObject/Service.php#L1962-L1972